### DISCUSSION TITLE
dynamically load vendor supplied scripts

### DISCUSSION THREAD

**Discussion by brianpmccullough** [SharePoint/sp-dev-docs#8579] [General]
### What type of issue is this?

Question

### What SharePoint development model, framework, SDK or API is this about?

💥 SharePoint Framework

### Target SharePoint environment

SharePoint Online

### What browser(s) / client(s) have you tested

- [ ] 💥 Internet Explorer
- [ ] 💥 Microsoft Edge
- [ ] 💥 Google Chrome
- [ ] 💥 FireFox
- [ ] 💥 Safari
- [ ] mobile (iOS/iPadOS)
- [ ] mobile (Android)
- [X] not applicable
- [ ] other (enter in the "Additional environment details" area below)

### Additional environment details

- Edge or Chrome latest/current (as of November 2022, but probably any)
- 15.2 (but probably any)
- 16.18.1 (but probably any)
- etc

### Issue description

I have Vendor supplied script(s).  The script should be loaded dynamically based on web part configuration.   Each vendor script has same "interface", however, slightly different implementations.  That is, i should be able to access the same props/methods in the vendor script namespace, but the js / script used changes. Preference to bundle script , rather than load from their CDN so that scripts can be reviewed "as-is". I put this together to show dynamic execution of script using require.context.  https://github.com/brianpmccullough/spfx-dynamic-require-sample

Thoughts?

**Comment by ghost** [comment]
Thank you for reporting this issue. We will be triaging your incoming issue as soon as possible.
